state-chain/chains/src/sol/api.rs

```diff
@@ -21,20 +22,36 @@ use crate::{
 
 use cf_primitives::{EgressId, ForeignChain};
 
+#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo)]
+pub struct TokenEnvironment {
+	pub token_mint_pubkey: SolAddress,
+	pub token_vault_ata: SolAddress,
+	pub token_vault_pda_account: SolAddress,
```

Comment:
The `token_vault_pda_account` is the same across tokens, only the `token_vault_pda_account` account changes on a per token basis. Not sure if you did that so we get all values in one lookup but I'd take it away as it's confusing.

Reply:
FYI. You wrote `token_vault_pda_account` twice. 

Maybe it would help if we have some doc comments explaining what each of these addresses are.

Reply:
ups, just fixed the typo, thanks.
It's documented in the Solana architecture doc, both explained and in the diagram.
https://www.notion.so/chainflip/Solana-Architecture-955481ad1afa468b8bb1a0af0897d18e?pvs=4#2626b37dac5e4cbb8312015e1759bc97

Reply:
will update